File: MaxElementLinkedList/Max.c
#include<stdio.h>
#include<stdlib.h>

struct Node 
{
    int data;
    struct Node * next;
};

typedef struct Node NODE;
typedef struct Node * PNODE;
typedef struct Node ** PPNODE;

void InsertFirst(PPNODE Head,int value)
{
    PNODE newn = (PNODE)malloc(sizeof(NODE));
    newn->data = value;
    newn->next = NULL;

    if(*Head == NULL)
    {
        *Head = newn;
    }
    else
    {
        newn->next = *Head;
        *Head = newn;
    }
}

void InsertLast(PPNODE Head,int value)
{
    PNODE newn = (PNODE)malloc(sizeof(NODE));
    newn->data = value;
    newn->next = NULL;
    PNODE temp = *Head;

    if(temp == NULL)
    {
        temp = newn;
    }
    else
    {
        while(temp->next != NULL)
        {
            temp = temp->next;
        }
        temp->next = newn;
    }
}

void Display(PNODE Head)
{
    while(Head != NULL)
    {
        printf("|%d|->",Head->data);
        Head = Head->next;
    }
}

int Count(PNODE Head)
{
    int icnt = 0;

    while(Head != NULL)
    {
        icnt++;
        Head = Head->next;
    }

    return icnt;
}

int MaxElement(PNODE Head)
{
    int Max = 0;

    Max = Head->data;
    while(Head != NULL)
    {
        if(Head->data > Max)
        {
            Max = Head->data;
        }

        Head = Head->next;
    }

    return Max;
}
int main()
{
    int iret = 0;
    PNODE First = NULL;

    printf("Jay Ganesh...\n");

    InsertFirst(&First,240);
    InsertFirst(&First,320);
    InsertFirst(&First,230);
    InsertFirst(&First,110);
    InsertLast(&First,5000);
    Display(First);

    iret = Count(First);
    printf("\nCount of Elements From linked list is : %d",iret);

    iret = MaxElement(First);
    printf("\nMax Element From linked list is : %d",iret);

    return 0;
}

File: MaxElementLinkedList/Max.go
package main
import "fmt"

type Node struct {
	data int;
	next * Node;
}

type NODE = Node;
type PNODE = * Node;
type PPNODE = ** Node;

func InsertFirst(Head PPNODE,value int) {
	var newn PNODE = new(NODE);
	newn.data = value;
	newn.next = nil;

	if(*Head == nil) {
		*Head  = newn;
	} else {
		newn.next = *Head;
		*Head = newn;
	}
}

func InsertLast(Head PPNODE,value int) {
	var newn PNODE = new(NODE);
	newn.data = value;
	newn.next = nil;
	var temp PNODE = *Head;

	if(temp == nil) {
		temp = newn;
	} else {
		for(temp.next != nil) {
			temp = temp.next;
		}
		temp.next = newn;
	}
}

func Display(Head PNODE) {
	for(Head != nil){
		fmt.Printf("|%d|->",Head.data);
		Head = Head.next;
	}
}

func Count(Head PNODE) int {
	var icnt int = 0;
	for(Head != nil){
		icnt++;
		Head = Head.next;
	}

	return icnt;
}

func MaxElement(Head PNODE) int {
	var iMax int = 0;

	iMax = Head.data;
	for(Head != nil){
		if(Head.data > iMax) {
			iMax = Head.data;
		}
		Head = Head.next;
	}

	return iMax;
}

func main() {
	var First PNODE = nil;
	var iret int = 0;

	fmt.Println("Jay Ganesh....");
	InsertFirst(&First,51);
	InsertFirst(&First,11);
	InsertFirst(&First,10);
	InsertLast(&First,105);
	InsertLast(&First,100);

	Display(First);
	iret = Count(First);
	fmt.Printf("\nCount of Elements From Linked list is : %d",iret);

	iret = MaxElement(First);
	fmt.Printf("\nMax Elements From Linked list is : %d",iret);

}